src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseUserServiceImpl.java

```diff
@@ -351,6 +358,35 @@ public U create(final @NotNull U user) throws PasswordValidationException {
         return rawUserRepository.saveAndFlush(user);
     }
 
+    @Override
+    public U createExternal(@NotNull U user, String origin) throws PasswordValidationException {
+
+        // External users are enabled by default
+        user.setEnabled(true);
+        // TODO setCountry and setStateProvince from setFields?
+        Assert.notNull(origin, "For external users, origin is mandatory");
+        user.setOrigin(origin);
+        setInitialTimestamps(user, new Date());
+        // TODO Need to make meaningful choice here
+        user.setProfessionalType(professionalTypeService.list().iterator().next());
```

Comment:
We have a predefined set of professional types. Why not to get one from the configuration?

Reply:
Not sure what you mean here, I only found that list is exists in DB.
UserImportService has a similar problem and it is not exactly solved there as well.
Frankly speaking, I think we should just consider making this field optional in DB.

Reply:
We are absolutely not getting it from configuration, as tests have different setup than normal flyway scripts. 
So iterator.next or random is as good as it gets, unless we make field optional